### DISCUSSION TITLE
Support for Cryptomator FS under Windows

### DISCUSSION THREAD

**Discussion by ghost** [photoprism/photoprism#4077] [Q&A]
---------------------------------------------------------------------------

#### 1. What is not working as documented?

It is impossible to use a Cryptomator container for the "originals" folder. When mounting a Cryptomator volume under Windows using WinFSP, then attempting to use that folder as the "originals" folder in photoprism (docker-compose.yml), the following message is generated: 
`Error response from daemon: The volume does not contain a recognized file system.
Please make sure that all required file system drivers are loaded and that the volume is not corrupted.`. Windows reports the FS of the drive as "cryptoFS".
When mounting the share using "WebDAV (File Explorer)", PhotoPrism doesn't throw any errors, but also finds no files in the folder and reports the directory as empty even after forcing a scan. This is all done with Docker running in Hyper-V mode.

#### 2. How can we reproduce it?

1. Install Photoprism for Windows using Docker
2. Switch Docker to Hyper-V, as per instructions
3. Create a Cryptomator container
4. Mount it with WinFsp or WebDAV (File Explorer)
5. Attempt to use it as the "originals" folder

#### 3. What behavior do you expect?

The folder is mounted successfully and all files inside are recognized

#### 4. What could be the cause of your problem?

Photoprism/Windows/Docker not having the necessary driver support for cryptoFS?

#### 5. Can you provide us with example files for testing, error logs, or screenshots?

No logs, as the software doesn't start with the expected configuration

#### 6. Which software versions do you use?

(a) PhotoPrism Architecture & Build Number: AMD64, build 231128-f48ff16ef

(b) Database Type & Version: MariaDB, nothing modified from the default

(c) Operating System Types & Versions: Windows 10 Pro 22H2

**Comment by lastzero** [comment]
Based on the feature description on the [WinFSP website](https://winfsp.dev/), I would guess that this is due to the filesystem implementation in Windows and/or how Docker interacts with it, e.g. userspace filesystems might not be supported (yet):

> WinFsp is system software that provides runtime and development support for custom file systems on Windows computers. In this sense it is similar to FUSE (Filesystem in Userspace), which provides the same functionality on UNIX-like computers.
